docs/pages/core-concepts.mdx

```diff
@@ -44,63 +45,91 @@ infrastructure from the public internet without the need for a VPN.
 
 Proxies TCP traffic in the native protocols of popular databases, including
-PostgreSQL and MySQL.
+PostgreSQL and MySQL. Use to enforce role-based access to databases with full audit logging.
 
 Read more about the [Teleport Database
 Service](./enroll-resources/database-access/database-access.mdx).
 
+### Teleport Discovery Service
+
+The Teleport Discovery Service automates the process of finding and enrolling cloud resources such as AWS EC2 instances and Azure VMs into a 
+Teleport cluster. It continuously scans supported cloud environments and adds matching resources based on configured rules, 
+reducing the need for manual node registration. 
+
+Read more about the [Teleport Discovery Service](./enroll-resources/auto-discovery/auto-discovery.mdx).
+
 ### Teleport Desktop Service
 
-Proxies Remote Desktop Protocol traffic to Windows desktops.
+Proxies Remote Desktop Protocol (RDP) traffic to Windows desktops. Enables access control and session recording for Windows systems
 
 Read more about the [Teleport Desktop
 Service](./enroll-resources/desktop-access/introduction.mdx).
 
-### Teleport Kubernetes Service
+### Teleport Jamf Service
 
-Proxies HTTP traffic to the Kubernetes API server.
+The Jamf Service integrates with Jamf to enrolls trusted devices for macOS endpoints. 
+Before a user is granted access, Teleport checks the device's compliance status with Jamf. 
+This service is ideal for organizations that need to verify macOS device posture as part of 
+their access control strategy, ensuring only trusted devices are allowed to connect. 
+[Read more about the Jamf Service](./identity-governance/device-trust/jamf-integration.mdx).
 
-Read more about the [Teleport Kubernetes
-Service](./enroll-resources/kubernetes-access/introduction.mdx)
+### Teleport Debug Service
 
-### Teleport SSH Service
+Provides internal diagnostics, exposing metrics, performance profiling tools, and debugging endpoints. It is designed for development and troubleshooting scenarios, enabling operators and developers to inspect the internal state of Teleport services. 
+This can be helpful for performance tuning, analyzing bottlenecks, or diagnosing unexpected behavior in a controlled environment. The debug service is enabled by default.
+[Read more about the Debug Service](./admin-guides/management/admin/troubleshooting.mdx#step-13-enable-verbose-logging).
 
-An SSH server implementation that allows users to execute commands on remote
-machines while taking advantage of Teleport's built-in access controls,
-auditing, and session recording.
+## Agent
 
-Read more about the [Teleport SSH Service](./enroll-resources/server-access/introduction.mdx).
+An instance of a **Teleport service** is called an **agent**. For all Teleport
+services besides the **Teleport SSH Service**, an agent can enable access to
+multiple resources, and can run on a separate host from the resources it enables
+access to. All agents must run in the same network as their target resources.
```

Comment:
This definition has several issues:
- according to it, an instance of the proxy service is an agent
- according to it, a single teleport process running multiple services is multiple agents. In such case we tend to refer to the teleport process as a whole as the agent. (this part is subjective, maybe it's only my perception of the term, you might want to check with other Teleporters before changing)

Potential agents definitions would be:
- a teleport instance running one or more services providing access to local resources
- a Teleport instance not running a proxy or auth service

I'm not sure if we should consider the jamf or okta services agents.